buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -66,13 +67,21 @@
     private static final String GRADLE_JDK_VERSION = "12.0.1+12@69cfe15208a647278a19ef0990eea691";
 
     // all distributions used by distro tests. this is temporary until tests are per distribution
-    private static final String PACKAGING_DISTRIBUTION = "packaging";
+    private static final String PACKAGING_CONFIGURATION = "packaging";
+    private static final String UPGRADE_CONFIGURATION = "packaging_upgrade";
+    private static final String PLUGINS_CONFIGURATION = "packaging_plugins";
     private static final String COPY_PACKAGING_TASK = "copyPackagingArchives";
+    private static final String COPY_UPGRADE_TASK = "copyUpgradeArchives";
+    private static final String COPY_PLUGINS_TASK = "copyPlugins";
     private static final String IN_VM_SYSPROP = "tests.inVM";
 
     private static Version upgradeVersion;
     private Provider<Directory> archivesDir;
+    private Provider<Directory> upgradeDir;
```

Comment:
I think we should initialize these providers early on in the `apply()` method. By doing it later we can very easily fuck things up by reordering when we call these various `configure*()` methods.